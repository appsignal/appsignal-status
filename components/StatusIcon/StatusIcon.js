import PropTypes from "prop-types";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import {
  faCheck,
  faExclamation,
  faInfo,
} from "@fortawesome/free-solid-svg-icons";

const iconMapping = {
  success: {
    icon: faCheck,
    style: "bg-green-500 text-white",
  },
  warning: {
    icon: faExclamation,
    style: "bg-orange-500 text-white",
  },
  error: {
    icon: faExclamation,
    style: "bg-red-500 text-white",
  },
  default: {
    icon: faInfo,
    style: "bg-gray-200 text-gray-700",
  },
};

const StatusIcon = ({ status }) => {
  const style = iconMapping[status].style;
  const icon = iconMapping[status].icon;

  return (
    <span
      className={`flex items-center justify-center h-4 w-4 rounded-full ${style}`}
    >
      <FontAwesomeIcon
        role="img"
        icon={icon}
        style={{ fontSize: "8px" }}
        fixedWidth
      />
    </span>
  );
};

StatusIcon.propTypes = {
  status: PropTypes.string.isRequired,
};

export default StatusIcon;
