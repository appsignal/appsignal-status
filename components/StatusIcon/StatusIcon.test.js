import { render, screen } from "@testing-library/react";

import StatusIcon from "./StatusIcon";

const build = (props = {}) => {
  return render(<StatusIcon status="success" {...props} />);
};

describe("StatusIcon", () => {
  test("it renders without problems", () => {
    const { container } = build();
    expect(container).toMatchSnapshot();
  });

  describe("different states", () => {
    test("default", () => {
      const { container } = build({ status: "default" });

      const icon = screen.getByRole("img", { hidden: true });
      expect(container).toMatchSnapshot();
      expect(icon.parentNode.classList).toContain("bg-gray-200");
      expect(icon.classList).toContain("fa-info");
    });

    test("success", () => {
      const { container } = build({ status: "success" });

      const icon = screen.getByRole("img", { hidden: true });
      expect(container).toMatchSnapshot();
      expect(icon.parentNode.classList).toContain("bg-green-500");
      expect(icon.classList).toContain("fa-check");
    });

    test("warning", () => {
      const { container } = build({ status: "warning" });

      const icon = screen.getByRole("img", { hidden: true });
      expect(container).toMatchSnapshot();
      expect(icon.parentNode.classList).toContain("bg-orange-500");
      expect(icon.classList).toContain("fa-exclamation");
    });

    test("error", () => {
      const { container } = build({ status: "error" });

      const icon = screen.getByRole("img", { hidden: true });
      expect(container).toMatchSnapshot();
      expect(icon.parentNode.classList).toContain("bg-red-500");
      expect(icon.classList).toContain("fa-exclamation");
    });
  });
});
