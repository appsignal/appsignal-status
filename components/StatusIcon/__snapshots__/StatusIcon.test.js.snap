// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusIcon different states default 1`] = `
<div>
  <span
    class="flex items-center justify-center h-4 w-4 rounded-full bg-gray-200 text-gray-700"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-info fa-w-6 fa-fw "
      data-icon="info"
      data-prefix="fas"
      focusable="false"
      role="img"
      style="font-size: 8px;"
      viewBox="0 0 192 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M20 424.229h20V279.771H20c-11.046 0-20-8.954-20-20V212c0-11.046 8.954-20 20-20h112c11.046 0 20 8.954 20 20v212.229h20c11.046 0 20 8.954 20 20V492c0 11.046-8.954 20-20 20H20c-11.046 0-20-8.954-20-20v-47.771c0-11.046 8.954-20 20-20zM96 0C56.235 0 24 32.235 24 72s32.235 72 72 72 72-32.235 72-72S135.764 0 96 0z"
        fill="currentColor"
      />
    </svg>
  </span>
</div>
`;

exports[`StatusIcon different states error 1`] = `
<div>
  <span
    class="flex items-center justify-center h-4 w-4 rounded-full bg-red-500 text-white"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-exclamation fa-w-6 fa-fw "
      data-icon="exclamation"
      data-prefix="fas"
      focusable="false"
      role="img"
      style="font-size: 8px;"
      viewBox="0 0 192 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M176 432c0 44.112-35.888 80-80 80s-80-35.888-80-80 35.888-80 80-80 80 35.888 80 80zM25.26 25.199l13.6 272C39.499 309.972 50.041 320 62.83 320h66.34c12.789 0 23.331-10.028 23.97-22.801l13.6-272C167.425 11.49 156.496 0 142.77 0H49.23C35.504 0 24.575 11.49 25.26 25.199z"
        fill="currentColor"
      />
    </svg>
  </span>
</div>
`;

exports[`StatusIcon different states success 1`] = `
<div>
  <span
    class="flex items-center justify-center h-4 w-4 rounded-full bg-green-500 text-white"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-check fa-w-16 fa-fw "
      data-icon="check"
      data-prefix="fas"
      focusable="false"
      role="img"
      style="font-size: 8px;"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
        fill="currentColor"
      />
    </svg>
  </span>
</div>
`;

exports[`StatusIcon different states warning 1`] = `
<div>
  <span
    class="flex items-center justify-center h-4 w-4 rounded-full bg-orange-500 text-white"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-exclamation fa-w-6 fa-fw "
      data-icon="exclamation"
      data-prefix="fas"
      focusable="false"
      role="img"
      style="font-size: 8px;"
      viewBox="0 0 192 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M176 432c0 44.112-35.888 80-80 80s-80-35.888-80-80 35.888-80 80-80 80 35.888 80 80zM25.26 25.199l13.6 272C39.499 309.972 50.041 320 62.83 320h66.34c12.789 0 23.331-10.028 23.97-22.801l13.6-272C167.425 11.49 156.496 0 142.77 0H49.23C35.504 0 24.575 11.49 25.26 25.199z"
        fill="currentColor"
      />
    </svg>
  </span>
</div>
`;

exports[`StatusIcon it renders without problems 1`] = `
<div>
  <span
    class="flex items-center justify-center h-4 w-4 rounded-full bg-green-500 text-white"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-check fa-w-16 fa-fw "
      data-icon="check"
      data-prefix="fas"
      focusable="false"
      role="img"
      style="font-size: 8px;"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
        fill="currentColor"
      />
    </svg>
  </span>
</div>
`;
